packages/table/components/Table.tsx

```diff
@@ -0,0 +1,172 @@
+    (data: T[]): Array<{}> | T[] => {
+      return [{}, ...data];
+    }
+  );
+
+  public getColumnSizes = memoizeOne(
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGridSize = this.updateGridSize.bind(this);
+  }
+
+  public render() {
+    const mergedData = this.getData(this.props.data);
+
+    const columnCount = React.Children.count(this.props.children);
+
+    return (
+      <AutoSizer
+        defaultWidth={1024}
+        defaultHeight={768}
+        onResize={this.updateGridSize}
+      >
+        {({ width, height }) => {
+          const columnSizes = this.getColumnSizes(
+            React.Children.toArray(this.props.children) as Array<
+              React.ReactElement<IColumnProps>
+            >,
+            width
+          );
+
+          const getColumnSize = ({ index }) => columnSizes[index];
+          return (
+            <MultiGrid
+              ref={this.setRef}
+              fixedColumnCount={1}
+              fixedRowCount={1}
+              cellRenderer={this.getCell}
+              columnWidth={getColumnSize}
+              columnCount={columnCount}
+              enableFixedColumnScroll={true}
+              enableFixedRowScroll={true}
+              height={height}
+              rowHeight={rowHeight}
+              rowCount={mergedData.length}
+              width={width}
+              hideTopRightGridScrollbar={true}
+              hideBottomLeftGridScrollbar={true}
+            />
+          );
+        }}
+      </AutoSizer>
+    );
+  }
+
+  private getHeaderCell({
+    key,
+    style,
+    column
+  }: {
+    style: any;
+    key: any;
+    column: React.ReactElement<IColumnProps>;
+  }) {
+    const className = css({
+      boxSizing: "border-box",
+      borderBottom: `1px solid ${Colors().black}`,
+      ...style
+    });
+    return <div {...{ className, key }}>{column.props.label}</div>;
+  }
+
+  private getCell(args) {
+    const { columnIndex, rowIndex } = args;
+    const data = this.getData(this.props.data);
+
+    const column = React.Children.toArray(this.props.children)[
+      columnIndex
+    ] as React.ReactElement<IColumnProps>;
+
+    if (!column || column.type !== Column) {
+      return null;
+    }
+
+    return rowIndex === 0
+      ? this.getHeaderCell({ ...args, column, data })
+      : this.getContentCell({ ...args, column, data });
+  }
+
+  private getContentCell({
```

Comment:
same, please consider not destructing arguments

Reply:
agree